python/src/scipp/plot/plot.py

```diff
@@ -18,16 +18,13 @@ def plot(input_data, collapse=None, projection=None, axes=None, color=None,
     from .plot_collapse import plot_collapse
     from .dispatch import dispatch
 
-    # Create a list of variables which will then be dispatched to correct
-    # plotting function.
-    # Search through the variables and group the 1D datasets that have
-    # the same coordinates and units.
-    # tobeplotted is a dict that holds four items:
-    # {number_of_dimensions, Dataset, axes, line_parameters}.
     tp = type(input_data)
-    if tp is sc.DataProxy or tp is sc.DataArray:
+    # Convert non-Datasets to temporary Dataset
+    if (tp is sc.DataProxy) or (tp is sc.DataArray) or \
+       (tp is sc.VariableProxy) or (tp is sc.Variable):
```

Comment:
Not an issue now, but this copy will become an issue once we have data arrays that are many Gbyte in size. Maybe open an issue so we do not forget to fix this in the future?